x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -39,70 +36,57 @@ import {
 } from './schemas/inference_schema';
-import type { PipelineDefinition } from '../../common/types/trained_models';
-import { type TrainedModelConfigResponse } from '../../common/types/trained_models';
-import { mlLog } from '../lib/log';
-import { forceQuerySchema } from './schemas/anomaly_detectors_schema';
-import { modelsProvider } from '../models/model_management';
 
 export function filterForEnabledFeatureModels<
   T extends TrainedModelConfigResponse | estypes.MlTrainedModelConfig
 >(models: T[], enabledFeatures: MlFeatures) {
   let filteredModels = models;
   if (enabledFeatures.nlp === false) {
-    filteredModels = filteredModels.filter((m) => m.model_type === 'tree_ensemble');
+    filteredModels = filteredModels.filter((m) => m.model_type !== TRAINED_MODEL_TYPE.PYTORCH);
   }
-
   if (enabledFeatures.dfa === false) {
-    filteredModels = filteredModels.filter((m) => m.model_type !== 'tree_ensemble');
+    filteredModels = filteredModels.filter(
+      (m) => m.model_type !== TRAINED_MODEL_TYPE.TREE_ENSEMBLE
+    );
   }
-
   return filteredModels;
 }
 
-export const populateInferenceServicesProvider = (client: IScopedClusterClient) => {
-  return async function populateInferenceServices(
-    trainedModels: TrainedModelConfigResponse[],
-    asInternal: boolean = false
-  ) {
-    const esClient = asInternal ? client.asInternalUser : client.asCurrentUser;
-
-    try {
-      // Check if model is used by an inference service
-      const { endpoints } = await esClient.transport.request<{
-        endpoints: InferenceAPIConfigResponse[];
-      }>({
-        method: 'GET',
-        path: `/_inference/_all`,
-      });
-
-      const inferenceAPIMap = groupBy(
-        endpoints,
-        (endpoint) => endpoint.service === 'elser' && endpoint.service_settings.model_id
-      );
-
-      for (const model of trainedModels) {
-        const inferenceApis = inferenceAPIMap[model.model_id];
-        model.hasInferenceServices = !!inferenceApis;
-        if (model.hasInferenceServices && !asInternal) {
-          model.inference_apis = inferenceApis;
-        }
-      }
-    } catch (e) {
-      if (!asInternal && e.statusCode === 403) {
-        // retry with internal user to get an indicator if models has associated inference services, without mentioning the names
-        await populateInferenceServices(trainedModels, true);
-      } else {
-        mlLog.error(e);
-      }
-    }
-  };
-};
-
 export function trainedModelsRoutes(
   { router, routeGuard, getEnabledFeatures }: RouteInitialization,
   cloud: CloudSetup
 ) {
+  router.versioned
+    .get({
+      path: `${ML_INTERNAL_BASE_PATH}/trained_models_list`,
+      access: 'internal',
+      security: {
+        authz: {
+          requiredPrivileges: ['ml:canGetTrainedModels'],
+        },
+      },
+      summary: 'Get trained models list',
+      description:
+        'Retrieves a complete list if trained models with stats, pipelines, and indices.',
```

Comment:
Typo! Should be `of` instead of `if`.

Reply:
fixed in https://github.com/elastic/kibana/pull/200977/commits/1c41a8415aa3851db2fe842c90d344e7a4c2484a